src/repo/store.go

```diff
@@ -289,49 +289,31 @@ func (s *Store) GetGithubRepos(ctx context.Context, userID string) ([]domain.Rep
 	return repos, nil
 }
 
-func (s *Store) ConnectRepo(ctx context.Context, userID, repoID string) (domain.Repository, error) {
-	query, args, err := s.sq.Update("installedRepos").
-		Set("connected", true).
-		Where(sq.Eq{"id": repoID, "userId": userID}).
-		Suffix("RETURNING id, githubId, fullName, private, defaultBranch,  status, connected").
-		ToSql()
-	if err != nil {
-		return domain.Repository{}, fmt.Errorf("failed to build ConnectRepoBranch query: %w", err)
-	}
-
-	row := s.db.QueryRowContext(ctx, query, args...)
-	if err != nil {
-		return domain.Repository{}, fmt.Errorf("failed to execute ConnectRepoBranch: %w", err)
+func (s *Store) ConnectRepo(ctx context.Context, userID, repoID, branch string) (domain.Repository, error) {
+	if branch == "" {
+		return domain.Repository{}, fmt.Errorf("branch cannot be empty")
 	}
-	var repo domain.Repository
-	if err := row.Scan(&repo.TreenqID, &repo.ID, &repo.FullName, &repo.Private, &repo.DefaultBranch, &repo.Status, &repo.Connected); err != nil {
-		if errors.Is(err, sql.ErrNoRows) {
-			return repo, domain.ErrRepoNotFound
-		}
-	}
-	return repo, nil
-}
 
-func (s *Store) UpdateRepoPrivateFlagAndDefaultBranch(ctx context.Context, newDefaultBranch string, isPrivate bool, repoID int) (domain.Repository, error) {
 	query, args, err := s.sq.Update("installedRepos").
-		Set("defaultBranch", newDefaultBranch).
-		Set("private", isPrivate).
-		Where(sq.Eq{"githubId": repoID}).
-		Suffix("RETURNING id, githubId, fullName, private, defaultBranch,  status, connected").
+		Set("connected", true).
```

Comment:
what do you think of making "connected" as a computed property? not in this PR, just an idea to think about. connected field might be as a function of a repo struct `return r.Branch != ""`